src/lib/actions/RulesAction.ts

```diff
@@ -44,7 +45,9 @@ export class RulesAction {
 			...enginePluginModules.map(pluginModule => core.dynamicallyAddEnginePlugin(pluginModule))
 		];
 		await Promise.all(addEnginePromises);
-		const selectOptions = input.workspace ? {workspace: await core.createWorkspace(input.workspace)} : undefined;
+		const selectOptions = input.workspace
+			? {workspace: await (new WorkspaceFactory()).create(core, input.workspace)}
```

Comment:
This feels off. If we wanted a true factory, then this probably would have been passed in and injected into the RulesAction class. But this really is just implementation detail... it seems just like a function to me.

So instead of calling this a factory, maybe just make this a function for now.

Otherwise, I would say, let's pass the CodeAnalyzer instance into the Factory's constructor and keep the create method lightweight so that this could be an interface. But all that advice is not needed. Right now, as the code is still young - I recommend we stick with functions until we know what classes we'll actually need. Let's avoid the "new" whenever possible.